packages/monorepo-builder/src/FileSystem/ComposerJsonProvider.php

```diff
@@ -62,9 +62,14 @@ public function getRootAndPackageFileInfos(): array
         );
     }
 
-    public function getPackageFileInfoByName(string $packageName): SmartFileInfo
+    /**
+     * @param SmartFileInfo[]|null $packageComposerFiles
+     */
+    public function getPackageFileInfoByName(string $packageName, ?array $packageComposerFiles = null): SmartFileInfo
     {
-        $packageComposerFiles = $this->packageComposerFinder->getPackageComposerFiles();
```

Comment:
The idea is good, just the implementation should happen inside this method.
Simple array memory property will do

Reply:
Ok done

Reply:
Thanks!

What about other calls of this method? Now some are cached and some are not

Reply:
Right. I just added the cache within `getPackagesComposerFileInfos()` too

Reply:
What about any future calls of `getPackageComposerFiles()`?

Reply:
So you want to move the caching from `ComposerJsonProvider` to `PackageComposerFinder`, right? Confirm and I'll do it

Reply:
:+1: That's the only reliable way I see. Can we do better?

Reply:
Ok done